src/com/hms_networks/americas/sc/localdatafiles/LocalDataFileManager.java

```diff
@@ -0,0 +1,476 @@
+   *     filesystem
+   */
+  public LocalDataFileManager(
+      String localDataFileRootFolder, int retainNumberMonths, long fileWriteIntervalMillis) {
+    this.localDataFileRootFolder = localDataFileRootFolder;
+    this.retainNumberMonths = retainNumberMonths;
+    this.fileWriteIntervalMillis = fileWriteIntervalMillis;
+    start();
+  }
+
+  /**
+   * Gets the file path of the data file for the current time.
+   *
+   * @return data file path
+   */
+  private String getCurrentDataFilePath() {
+    // Get current date/time
+    Calendar calendar = Calendar.getInstance();
+    long localTimeMillis =
+        System.currentTimeMillis() - LocalTimeOffsetCalculator.getLocalTimeOffsetMilliseconds();
+    calendar.setTimeInMillis(localTimeMillis);
+
+    // Get date sections (add 1 to month - based on 0)
+    String year = String.valueOf(calendar.get(Calendar.YEAR));
+    String month = String.valueOf(calendar.get(Calendar.MONTH) + 1);
+    String day = String.valueOf(calendar.get(Calendar.DAY_OF_MONTH));
+    String hour = String.valueOf(calendar.get(Calendar.HOUR));
+
+    // Pad month, day and hour with leading zero if only one digit
+    month = month.length() == 1 ? "0" + month : month;
+    day = day.length() == 1 ? "0" + day : day;
+    hour = hour.length() == 1 ? "0" + hour : hour;
+
+    return localDataFileRootFolder
+        + "/"
+        + year
+        + "/"
+        + month
+        + "/"
+        + day
+        + "/"
+        + LocalDataFileConstants.LOCAL_DATA_FILE_NAME_PREFIX
+        + year
+        + month
+        + day
+        + hour
+        + LocalDataFileConstants.LOCAL_DATA_FILE_EXTENSION;
+  }
+
+  /**
+   * Stops local data file management and returns when the local data file management thread has
+   * terminated.
+   *
+   * @throws InterruptedException if unable to wait for local data file management thread to
+   *     terminates
+   */
+  public void stopLocalDataFileManagement() throws InterruptedException {
+    doProcessing = false;
+    join();
+  }
+
+  /**
+   * Deletes outdated local data files from the file system. The configured value of {@link
+   * #retainNumberMonths} is subtracted from the current month to get the first month which will be
+   * deleted.
+   */
+  private void deleteOutdatedFiles() {
+    // Get current date/time
+    Calendar calendar = Calendar.getInstance();
+    long localTimeMillis =
+        System.currentTimeMillis() - LocalTimeOffsetCalculator.getLocalTimeOffsetMilliseconds();
+    calendar.setTimeInMillis(localTimeMillis);
+
+    // Get date sections (add 1 to month - based on 0)
+    int year = calendar.get(Calendar.YEAR);
+    int month = calendar.get(Calendar.MONTH) + 1;
+
+    /*
+     * Subtract AZConnectorConsts.LOCAL_DATA_FILE_DELETE_AFTER_MONTHS from current date to get month to begin deleting.
+     */
+    for (int x = 0; x <= retainNumberMonths; x++) {
+      month--;
+      if (month == 0) {
+        month = 12;
```

Comment:
there is no constant you can use for this?

Reply:
no native Java constant that i know of or could find. I made a constant now though